Modelica_DeviceDrivers/Resources/Include/MDDUDPSocket.h

```diff
@@ -296,21 +318,23 @@ void* MDD_udpReceivingThread(void * p_udp) {
                     if (udp->nReceivedBytes > 0) (udp->nRecvbufOverwrites)++;
                     /* Receive the next datagram  */
                     udp->nReceivedBytes =
-                        recvfrom(udp->sock,                   /* UDP socket */
+                        recvfrom(udp->sock,                    /* UDP socket */
                                  udp->msgInternal,             /* receive buffer */
                                  udp->messageLength,           /* max bytes to receive */
                                  0,                            /* no special flags */
                                  (struct sockaddr*) &(udp->sa),/* address of sender */
                                  &sa_len
                                 );
-                    pthread_mutex_unlock(&(udp->messageMutex));
-
                     if (udp->nReceivedBytes < 0) {
+                        pthread_mutex_unlock(&(udp->messageMutex));
                         ModelicaFormatError("MDDUDPSocket.h: recvfrom(..) failed (%s)\n",
                                             strerror(errno));
                     }
-                    break;
+                    else {
+                        pthread_mutex_unlock(&(udp->messageMutex));
+                    }
                 }
+                break;
```

Comment:
OK, nicer to have the break outside the else branch (though the only possibility that the else branch is not called is that an error occured and `ModelicaFormatError` aborts the simulation anyway).